src/Asm89/Stack/CorsService.php

```diff
@@ -74,16 +74,18 @@ public function isPreflightRequest(Request $request)
 
     public function addActualRequestHeaders(Response $response, Request $request)
     {
-        if (!$this->checkOrigin($request)) {
-            return $response;
-        }
-
-        $response->headers->set('Access-Control-Allow-Origin', $request->headers->get('Origin'));
+        if ($this->options['supportsCredentials'] || !empty($this->options['allowedOriginsPatterns'])) {
+            if ($this->checkOrigin($request)) {
```

Comment:
If I have `supportsCredentials` set to `true` but `allowedOrigins` set to `[]` there is no need to add this and no need to vary the request by `Origin`. Adding `Access-Control-Allow-Credentials: true` to all requests is fine since it wont pass the origin requirement in the browser anyways.

Reply:
If supportsCredentials or allowedOriginsPatterns we MUST add the actual Origin header as allowed, because patterns or not valid as header and the wildcard is not allowed when supportsCredentials is used.

Reply:
Right, I'm saying what happens if `supportsCredentials` is `true` but `allowedOriginsPatterns` is empty?
What if `allowedOrigins` is **not** empty?

imho, `supportsCredentials` should add it's header, regardless of the other settings.

Reply:
Ah I guess when it's not a wildcard, we could just add the headers indeed.